src/flow/darcy_flow_mh.hh

```diff
@@ -188,6 +194,8 @@ public:
 	Field<3, FieldValue<3>::Scalar> field_ele_piezo_head;
         Field<3, FieldValue<3>::VectorFixed > field_ele_velocity;
 
+        BCField<3, FieldValue<3>::Enum > nonlinear_solver;
```

Comment:
It does not make sense to define nonlinear_solver as BCField in EqData. It should be just a key in the input record of DarcyMH.